public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -38,6 +53,34 @@ export function parseMatcher(matcher: string): Matcher {
   };
 }
 
+/**
+ * This function combines parseMatcher and parsePromQLStyleMatcher, always returning an array of Matcher[] regardless of input syntax
+ */
+export function parseMatcherToArray(matcher: string): Matcher[] {
+  return isPromQLStyleMatcher(matcher) ? parsePromQLStyleMatcher(matcher) : [parseMatcher(matcher)];
+}
+
+/**
+ * This function turns a PromQL-style matchers like { foo="bar", bar!=baz } in to an array of Matchers
+ */
+export function parsePromQLStyleMatcher(matcher: string): Matcher[] {
+  if (!isPromQLStyleMatcher(matcher)) {
+    throw new Error('not a PromQL style matcher');
+  }
+
+  // split by `,` but not when it's used as a label value
+  const commaUnlessQuoted = /,(?=(?:[^"]*"[^"]*")*[^"]*$)/;
+  const parts = matcher.replace(/^\{/, '').replace(/\}$/, '').trim().split(commaUnlessQuoted);
+
+  return compact(parts)
+    .flatMap(parseMatcher)
+    .map((matcher) => ({
+      ...matcher,
+      name: unquoteWithUnescape(matcher.name),
+      value: unquoteWithUnescape(matcher.value),
```

Comment:
I'm not sure we should `unescapeWithUnquote` in this place. The `parseMatchers` function doesn't do that.
What happens if you try to save a policy tree containing promql matchers with quotes and escapes? Will it remove them from all matchers in the tree?

Reply:
I don't think it would, since we seem to always apply `quoteWithEscape` to the matcher name and value when we send the payload the the Alertmanager config API.

https://github.com/grafana/grafana/blob/926260268a3904bf97aea82c04ef06f188d04123/public/app/features/alerting/unified/utils/amroutes.ts#L192-L194

Reply:
But your point on `parseMatcher` is definitely valid, if that one doesn't decode then neither should this one.